tests/lib/rules/yoda.js

```diff
@@ -992,6 +1017,110 @@ ruleTester.run("yoda", rule, {
                     type: "BinaryExpression"
                 }
             ]
+        },
+        {
+            code: "if (`green` < x.y && x.y < `blue`) {}",
+            output: "if (`green` < x.y && `blue` > x.y) {}",
+            options: ["always", { exceptRange: true }],
+            parserOptions: { ecmaVersion: 2015 },
+            errors: [
+                {
+                    messageId: "expected",
+                    data: { expectedSide: "left", operator: "<" },
+                    type: "BinaryExpression"
+                }
+            ]
+        },
+        {
+            code: "if (x < -1n || 1n <= x) {}",
+            output: "if (x < -1n || x >= 1n) {}",
+            options: ["never", { exceptRange: true }],
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [
+                {
+                    messageId: "expected",
+                    data: { expectedSide: "right", operator: "<=" },
+                    type: "BinaryExpression"
+                }
+            ]
+        },
+        {
+            code: "if('a' <= x && x < 'b') {}",
```

Comment:
ok this is a false negative. It should be allowed. 

I will take a look in this 👍

Reply:
this is invalid currently as well in master 

[`demo`](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHlvZGE6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbmlmKCdhJyA8PSB4ICYmIHggPCAnYicpIHt9Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)